lms/models/lti_user.py

```diff
@@ -58,6 +59,21 @@ def from_auth_params(application_instance, lti_core_schema):
             email=lti_core_schema["lis_person_contact_email_primary"],
         )
 
+    def serialize(self) -> dict:
```

Comment:
Replaces `namedtuple`._asdict.

We don't want to have any general dataclass->dict conversion (dataclass.asdict for example) but rather have control of what gets seralized to be sent over the wire.

Reply:
I'd like to hear more about this. `dataclass.asdict` would seem to do exactly what we want for now. If we had a test which locked this down, is there a risk I'm missing?

Reply:
This started with me trying to have a "lti_roles: LTIRoles" property here which was complicated with namedtupled and the BearerToken mess.

With the dataclass we can add new fields no problem, but if we use `asdict` it's tricker to control what get serialized and how. A dumb method seems simpler.